### PR TITLE
[SYCL] Make queue::device_has() private

### PR DESCRIPTION
**Pull request by s-kanaev**
The function is not defined by the specification, and was marked as public by mistake.

### CODE REVIEW

**Review by alexbatashev — DISMISSED** [review]

File: sycl/include/CL/sycl/queue.hpp [hunk 1]
```diff
@@ -237,6 +237,7 @@ class __SYCL_EXPORT queue {
   template <info::queue param>
   typename info::param_traits<info::queue, param>::return_type get_info() const;
 
+private:
```
Comment (bader) [inline sycl/include/CL/sycl/queue.hpp:240]: If it doesn't break any test, I will be surprised.

Comment (alexbatashev) [reply sycl/include/CL/sycl/queue.hpp:240]: @bader this function is not on the spec, and it was introduced to be used internally. I don't think there're any tests for this function.

Comment (s-kanaev) [reply sycl/include/CL/sycl/queue.hpp:240]: This method is only used internally.

Comment (bader) [reply sycl/include/CL/sycl/queue.hpp:240]: This is a violation of the contribution process. See https://github.com/intel/llvm/blob/sycl/CONTRIBUTING.md#tests-development: 
> Every product change should be accompanied with corresponding test modification (adding new test(s), extending, removing or modifying existing test(s)).

Please, make sure that code contributed to the project is covered by some tests.

Comment (bader) [reply sycl/include/CL/sycl/queue.hpp:240]: > This method is only used internally.

You can't claim that for public API. It's used in SYCL-CTS - https://github.com/KhronosGroup/SYCL-CTS/pull/287#discussion_r814092413.

Comment (bader) [reply sycl/include/CL/sycl/queue.hpp:240]: http://ci.llvm.intel.com:8010/#builders/18/builds/21660 - that's what I expected. Although Linux testing results are "surprising".

**Comment by s-kanaev** [comment]
Should cmake version be bumped as windows symbols have changed?

File: sycl/include/CL/sycl/queue.hpp (hunk 1 above)
Comment (s-kanaev) [reply sycl/include/CL/sycl/queue.hpp:240]: On the contrary, I didn't expect failure on Windows platform. I didn't know they mangle access modifier in symbol name.

**Comment by alexbatashev** [comment]
> Should cmake version be bumped as windows symbols have changed?

No

**Review by gmlueck — APPROVED** [review]

**Review by alexbatashev — APPROVED** [review]

**Pull request merged**